### DISCUSSION TITLE
When is triggered the Signal Detected event?

### DISCUSSION THREAD

**Discussion by Transcan** [EDCD/EDDI#2290] [Needs thought]
Hello there,

What I'm concerned about is if Signal Detected event is triggered by FSS.
My goal is to create a script that informs me whenever I scan a planet that has biology on it (I'm talking about Odyssey surface signals).
That information is shown in the upper right part of the FSS screen, right to the table of material percentage the body has, but I always forget to check it and quickly scan the whole system.
So that EDDI warn would be great.

I wasn't sure if that was shown in the journal, but it is:
In this Journal, I just finished scanning a planet (in FSS) that has 4 biology signals on it:
```
{ "timestamp":"2021-11-22T23:03:23Z", "event":"FSSBodySignals", "BodyName":"Outordy SS-D b53-0 3 a", "BodyID":4, "SystemAddress":625588904905, "Signals":[ { "Type":"$SAA_SignalType_Biological;", "Type_Localised":"Biológica", "Count":1 } ] }
{ "timestamp":"2021-11-22T23:03:24Z", "event":"Scan", "ScanType":"Detailed", "BodyName":"Outordy SS-D b53-0 3 a", "BodyID":4, "Parents":[ {"Planet":3}, {"Star":0} ], "StarSystem":"Outordy SS-D b53-0", "SystemAddress":625588904905, "DistanceFromArrivalLS":1663.314123, "TidalLock":false, "TerraformState":"", "PlanetClass":"Icy body", "Atmosphere":"thin argon atmosphere", "AtmosphereType":"Argon", "AtmosphereComposition":[ { "Name":"Argon", "Percent":100.000000 } ], "Volcanism":"", "MassEM":0.001121, "Radius":927268.500000, "SurfaceGravity":0.519687, "SurfaceTemperature":52.383320, "SurfacePressure":144.670776, "Landable":true, "Materials":[ { "Name":"sulphur", "Name_Localised":"Azufre", "Percent":27.200178 }, { "Name":"carbon", "Name_Localised":"Carbono", "Percent":22.872532 }, { "Name":"phosphorus", "Name_Localised":"Fósforo", "Percent":14.643388 }, { "Name":"iron", "Name_Localised":"Hierro", "Percent":12.057595 }, { "Name":"nickel", "Name_Localised":"Níquel", "Percent":9.119860 }, { "Name":"manganese", "Name_Localised":"Manganeso", "Percent":4.979664 }, { "Name":"selenium", "Name_Localised":"Selenio", "Percent":4.257058 }, { "Name":"vanadium", "Name_Localised":"Vanadio", "Percent":2.960926 }, { "Name":"yttrium", "Name_Localised":"Ytrio", "Percent":0.720186 }, { "Name":"tungsten", "Name_Localised":"Tungsteno", "Percent":0.662083 }, { "Name":"mercury", "Name_Localised":"Mercurio", "Percent":0.526535 } ], "Composition":{ "Ice":0.878254, "Rock":0.104374, "Metal":0.017372 }, "SemiMajorAxis":475896126.031876, "Eccentricity":0.000000, "OrbitalInclination":0.269956, "Periapsis":358.015372, "OrbitalPeriod":29053104.519844, "AscendingNode":61.314465, "MeanAnomaly":233.110969, "RotationPeriod":131773.103603, "AxialTilt":0.029745, "WasDiscovered":true, "WasMapped":false }
```
The problem is that I checked and the Signal Detected script wasn't fired up (and I guess it should).
And I don't know how to get that info from EDDI and use it in the scripts. There is no body parameter about its signals.

I see these types of signals are present in SignalSource.resx and they are triggered when you finish the DSS Scan, but not when I need them (in FSS).
Its because its not implemented yet or is because a bug?
Is there a way to do what I want in the current version of EDDI?
Thanks in advance.

**Comment by Tkael** [comment]
`FSSBodySignals` is actually a recent addition to the signal data that we're now receiving. EDDI doesn't support it yet but I'll take this query as interest in implementing the event.

**Reply by Transcan** [reply]
Oh, so I guessed right... =)
I will look forward to that implementation. It will give an even better Elite experience.

Just curious... How is going to be implemented? As and addition to the actual event? or maybe add a new property to body (something like an array _body.signal_ or similar)?
